src/System.Memory/tests/TestHelpers.cs

```diff
@@ -30,6 +30,12 @@ public static void ValidateReferenceType<T>(this Span<T> span, params T[] expect
             AssertThrows<IndexOutOfRangeException, T>(span, (_span) => ignore = _span[expected.Length]);
         }
 
+        public static unsafe void ValidateNonNullEmpty<T>(this Span<T> span)
```

Comment:
Why do we need this? Wouldn't the existing Validate\<T\> be sufficient to do both those checks?
If params T[] has length 0 (no params passed), then SequenceEqual effectively does the IsEmpty check.

Also, when would `Unsafe.AsPointer(ref MemoryMarshal.GetReference(span)) != null` fail?

Reply:
The second check is validating that empty Spans are not normalized to null. SequenceEqual is not validating that.